internal/controlplane/handlers_oauth.go

```diff
@@ -470,6 +473,48 @@ func (s *Server) VerifyProviderTokenFrom(ctx context.Context,
 	return &pb.VerifyProviderTokenFromResponse{Status: "OK"}, nil
 }
 
+// VerifyProviderCredential verifies the provider credential has been created for the matching enrollment nonce
+func (s *Server) VerifyProviderCredential(ctx context.Context,
+	in *pb.VerifyProviderCredentialRequest) (*pb.VerifyProviderCredentialResponse, error) {
+	entityCtx := engine.EntityFromContext(ctx)
+	projectID := entityCtx.Project.ID
+
+	enrollmentNonce := in.EnrollmentNonce
+
+	// Telemetry logging
+	logger.BusinessRecord(ctx).Project = projectID
+
+	// check if a token has been created matching the nonce
+	_, err := s.store.GetAccessTokenByEnrollmentNonce(ctx, db.GetAccessTokenByEnrollmentNonceParams{
+		ProjectID:       projectID,
+		EnrollmentNonce: sql.NullString{Valid: true, String: enrollmentNonce},
+	})
+
+	if err != nil {
+		if !errors.Is(err, sql.ErrNoRows) {
+			return nil, status.Errorf(codes.Internal, "error getting access token: %v", err)
+		}
```

Comment:
This might read more clearly as:

```suggestion
	if !errors.Is(err, sql.ErrNoRows) {
		if err == nil {
			return &pb.VerifyProviderCredentialResponse{Status:"OK"}, nil
		} else {
			return nil, status.Errorf(codes.Internal, "error getting access token: %v", err)
		}
	}
```

And just early-exit if there's a PAT row or a hard error.  Then the second search could also early-exit if it finds success, and the default case would be to return "KO".